openedx/core/djangoapps/dark_lang/views.py

```diff
@@ -2,164 +2,112 @@
 Views file for the Darklang Django App
 """
 from django.contrib.auth.decorators import login_required
+from django.http import Http404
+from django.shortcuts import redirect
+from django.template.loader import render_to_string
 from django.utils.decorators import method_decorator
 from django.utils.translation import ugettext as _
 from django.utils.translation import LANGUAGE_SESSION_KEY
-from django.views.generic.base import View
 
-from edxmako.shortcuts import render_to_response
 from openedx.core.djangoapps.dark_lang import DARK_LANGUAGE_KEY
 from openedx.core.djangoapps.dark_lang.models import DarkLangConfig
 from openedx.core.djangoapps.lang_pref import LANGUAGE_KEY
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
 from openedx.core.djangoapps.user_api.preferences.api import (
     delete_user_preference,
     get_user_preference,
     set_user_preference
 )
-from openedx.core.lib.api.view_utils import view_auth_classes
+from openedx.core.djangoapps.util.user_messages import PageLevelMessages
+from web_fragments.fragment import Fragment
 
-LANGUAGE_INPUT_FIELD = 'preview_lang'
+LANGUAGE_INPUT_FIELD = 'preview_language'
 
 
-@view_auth_classes()
-class DarkLangView(View):
+class PreviewLanguageFragmentView(EdxFragmentView):
     """
     View used when a user is attempting to change the preview language using Darklang.
 
     Expected Behavior:
     GET - returns a form for setting/resetting the user's dark language
     POST - updates or clears the setting to the given dark language
     """
-    template_name = 'darklang/preview_lang.html'
 
-    @method_decorator(login_required)
-    def get(self, request):
+    def render_to_fragment(self, request, course_id=None, **kwargs):
         """
-        Returns the Form for setting/resetting a User's dark language setting
-
-        Arguments:
-            request (Request): The Django Request Object
+        Renders the theming administration view as a fragment.
```

Comment:
Thanks, I did a search which just matched the two you found. I've fixed them both.